### PR TITLE
Create notebook with BigQuery examples

### PR DESCRIPTION
**Pull request by emmarogge**
Notebook contains examples and best practices for working with BigQuery resources in a VWB cloud env.

### CODE REVIEW

File: first_hour_on_vwb/working_with_data.ipynb [hunk 1]
```diff
@@ -0,0 +1,589 @@
...
+   },
+   "source": [
+    "### Objective\n",
+    "\n",
+    "Use this notebook to perform common workspace resource operations including:\n",
+    "\n",
+    "- [Access data from referenced resources](#access-ref-data)\n",
+    "  - [Read data from BigQuery](#read-from-bq)\n",
+    "    - [Using pandas-gbq library](#using-pandas)\n",
+    "    - [Using google-cloud-bigquery](#using-gcb)\n",
```
Comment (wnojopra) [inline first_hour_on_vwb/working_with_data.ipynb]: Double check these links, I think the read-from-bq and using-gcb links need to be fixed

File: first_hour_on_vwb/working_with_data.ipynb [hunk 2]
```diff
@@ -0,0 +1,589 @@
...
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Working with data from referenced resources\n",
+    "\n",
+    "<a id='access-ref-data'></a>\n",
+    "\n",
+    "A [referenced resource](https://terra-docs.api.verily.com/docs/getting_started/web_ui/#referenced-vs-workspace-controlled-resources) points to a source outside of the current workspace in order to represent data or other elements in Verily Workbench. To add a referenced resource to your workspace, use the <a href=\"../working_with_resources.ipynb\">working_with_resources.ipynb</a> notebook.\n",
```
Comment (wnojopra) [inline first_hour_on_vwb/working_with_data.ipynb]:
```suggestion
    "[Referenced resources](https://terra-docs.api.verily.com/docs/getting_started/web_ui/#referenced-vs-workspace-controlled-resources) represent data or other elements in Verily Workbench by pointing to a source that exists outside of the current workspace. To add a referenced resource to your workspace, use the <a href=\"../working_with_resources.ipynb\">working_with_resources.ipynb</a> notebook.\n",
```
Just to keep consistent with whats in the doc.

File: first_hour_on_vwb/working_with_data.ipynb [hunk 3]
```diff
@@ -0,0 +1,589 @@
...
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Query with cell magic\n",
+    "\n",
+    "To quickly to run a query, consider using the [IPython magics for BigQuery](https://cloud.google.com/python/docs/reference/bigquery/latest/magics). Several examples are provided below. This option is useful for troubleshooting, demos and developing new queries, although it's not particularly useful for preserving and comparing longitudinal results of queries you anticipate running routinely.\n",
```
Comment (wnojopra) [inline first_hour_on_vwb/working_with_data.ipynb]:
```suggestion
    "To quickly to run a query, we provide several examples below using [IPython magics for BigQuery](https://cloud.google.com/python/docs/reference/bigquery/latest/magics). This option is useful for troubleshooting, demos and developing new queries.\n",
```
Just trying to get the necessary points across. If you feel strongly about preserving and comparing longitudinal results, we should point to examples that help with that.

File: first_hour_on_vwb/working_with_data.ipynb [hunk 4]
```diff
@@ -0,0 +1,589 @@
...
+    "                self.resolve_button,\n",
+    "                self.output\n",
+    "            ],\n",
+    "            layout=wu.vbox_layout\n",
+    "        )\n",
+    "\n",
+    "    def get_bq_resources(self):\n",
+    "        result = subprocess.run(\n",
+    "            [\"terra\", \"resource\", \"list\", \"--format=JSON\"], capture_output=True, text=True)\n",
+    "        ids_list = wu.list_bq_tables(result.stdout)\n",
```
Comment (wnojopra) [inline first_hour_on_vwb/working_with_data.ipynb]: I'm not seeing the list_bq_tables function. Did you still need to check in those changes?

**Review by wnojopra — CHANGES_REQUESTED** [review]

File: first_hour_on_vwb/working_with_data.ipynb (hunk 1 above)
Comment (emmarogge) [reply first_hour_on_vwb/working_with_data.ipynb]: fixed!

File: first_hour_on_vwb/working_with_data.ipynb (hunk 2 above)
Comment (emmarogge) [reply first_hour_on_vwb/working_with_data.ipynb]: fixed!

File: first_hour_on_vwb/working_with_data.ipynb (hunk 4 above)
Comment (emmarogge) [reply first_hour_on_vwb/working_with_data.ipynb]: fixed!

File: first_hour_on_vwb/working_with_data.ipynb [hunk 5]
```diff
@@ -0,0 +1,577 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "%bigquery_stats bigquery-public-data.human_genome_variants.1000_genomes_pedigree"
```
Comment (wnojopra) [inline first_hour_on_vwb/working_with_data.ipynb]: Cool, I wasn't aware of bigquery_stats. Do you know if this is a costly query to run? I notice that it takes significantly longer than some of the other queries to run.

File: first_hour_on_vwb/working_with_data.ipynb [hunk 6]
```diff
@@ -0,0 +1,577 @@
...
+   },
+   "outputs": [],
+   "source": [
+    "%%bigquery\n",
+    "\n",
+    "SELECT\n",
+    "    COUNT(DISTINCT Maternal_ID) as num_mothers,\n",
+    "    COUNT(DISTINCT Paternal_ID) as num_fathers,\n",
+    "    COUNT(DISTINCT Individual_ID) as num_individuals,\n",
+    "FROM `bigquery-public-data.`;"
```
Comment (wnojopra) [inline first_hour_on_vwb/working_with_data.ipynb]: This doesn't work - it looks like you're missing the dataset/table

File: first_hour_on_vwb/working_with_data.ipynb [hunk 7]
```diff
@@ -0,0 +1,577 @@
...
+    "            print(self.query_result)\n",
+    "\n",
+    "    def resolve_resource(self, b):\n",
+    "        with self.output:\n",
+    "            commandList = [\"terra\", \"resolve\",\n",
+    "                           f\"--name={self.name_dropdown.value}\"]\n",
+    "\n",
+    "            print('Running command:')\n",
+    "            print('\\n'.join(commandList))\n",
+    "            print('')\n",
```
Comment (wnojopra) [inline first_hour_on_vwb/working_with_data.ipynb]: image
This outputs the commands one word per line.
Is it possible to have it so the output looks like
```
terra resolve \
  --name=bigquery-public-data.human_genome_variants.1000_genomes_pedigree
```

File: first_hour_on_vwb/README.md [hunk 8]
```diff
@@ -7,6 +7,8 @@ New users who engage with this workspace will be exposed to a variety of helpful
 
 ## Getting started
 
+- Run the [working_with_data.ipynb](https://github.com/DataBiosphere/terra-axon-examples/blob/main/first_hour_on_vwb/working_with_data.ipynb) notebook to learn about various options for interacting with BigQuery data from a [data collection](https://support.workbench.verily.com/docs/how_to_guides/work_with_data/#add-a-data-collection) in your workspace.
```
Comment (wnojopra) [inline first_hour_on_vwb/README.md]: My take is less of working with "data collection" and more accurately working with "referenced data"
```suggestion
- Run the [working_with_data.ipynb](https://github.com/DataBiosphere/terra-axon-examples/blob/main/first_hour_on_vwb/working_with_data.ipynb) notebook to learn about various options for interacting with BigQuery data from [referenced resources](https://support.workbench.verily.com/docs/how_to_guides/work_with_data/#add-a-data-reference) in your workspace.
```

File: first_hour_on_vwb/working_with_data.ipynb [hunk 9]
```diff
@@ -0,0 +1,577 @@
+{
```
Comment (wnojopra) [inline first_hour_on_vwb/working_with_data.ipynb]: The word "Data" can mean a lot of things so I wonder if we should be a bit more accurate with the name of this notebook. `working_with_data_references`? `working_with_referenced_resource_data`? `working_with_bq_resources`?

**Review by wnojopra — CHANGES_REQUESTED** [review]

File: first_hour_on_vwb/working_with_data.ipynb (hunk 7 above)
Comment (emmarogge) [reply first_hour_on_vwb/working_with_data.ipynb]: updated to look like [this](https://screenshot.googleplex.com/59at8kk6nmhem9K)

File: first_hour_on_vwb/working_with_data.ipynb (hunk 9 above)
Comment (emmarogge) [reply first_hour_on_vwb/working_with_data.ipynb]: renamed!

File: first_hour_on_vwb/README.md (hunk 8 above)
Comment (emmarogge) [reply first_hour_on_vwb/README.md]: fixed!

File: first_hour_on_vwb/working_with_data.ipynb (hunk 6 above)
Comment (emmarogge) [reply first_hour_on_vwb/working_with_data.ipynb]: fixed!

File: first_hour_on_vwb/working_with_data.ipynb (hunk 5 above)
Comment (emmarogge) [reply first_hour_on_vwb/working_with_data.ipynb]: I'm not exactly sure how much it costs. Looking at the [logs](https://screenshot.googleplex.com/4Tw7KH4JiEtuwAR), this query takes about 50 seconds to run, so it does take longer than the other queries in the notebook. However, it is running several queries per column (finding # observations, # distinct, # missing, % distinct, % missing, and plotting the frequency of each possible value) for each of the 12 columns in this particular table, I think it's not necessarily unreasonable that it takes this long. I could include a caveat that running this cell may take about a minute and could be expensive relative to other queries. What do you think?

File: first_hour_on_vwb/working_with_bq_resources.ipynb [hunk 10]
```diff
@@ -0,0 +1,579 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Working with data\n",
```
Comment (wnojopra) [inline first_hour_on_vwb/working_with_bq_resources.ipynb]: Title needs to be changed now

File: first_hour_on_vwb/working_with_data.ipynb (hunk 5 above)
Comment (wnojopra) [reply first_hour_on_vwb/working_with_data.ipynb]: Can we at least confirm and document the size of the dataset?

**Review by wnojopra — CHANGES_REQUESTED** [review]

**Review by wnojopra — APPROVED** [review]

**Pull request merged**